sklearn/compose/tests/test_column_transformer.py

```diff
@@ -208,24 +213,27 @@ def test_column_transformer_dataframe():
     assert_array_equal(both.fit_transform(X_df), res)
     assert_array_equal(both.fit(X_df).transform(X_df), res)
     assert len(both.transformers_) == 2
-    assert ct.transformers_[-1][0] != 'remainder'
+    assert both.transformers_[-1][0] != 'remainder'
+    assert both.transformers_output_ == {'trans1': [0], 'trans2': [1]}
 
     # test multiple columns
     both = ColumnTransformer([('trans', Trans(), ['first', 'second'])],
                              transformer_weights={'trans': .1})
     assert_array_equal(both.fit_transform(X_df), 0.1 * X_res_both)
     assert_array_equal(both.fit(X_df).transform(X_df), 0.1 * X_res_both)
     assert len(both.transformers_) == 1
-    assert ct.transformers_[-1][0] != 'remainder'
+    assert both.transformers_[-1][0] != 'remainder'
+    assert both.transformers_output_ == {'trans': [0, 1]}
 
     both = ColumnTransformer([('trans', Trans(), [0, 1])],
                              transformer_weights={'trans': .1})
     assert_array_equal(both.fit_transform(X_df), 0.1 * X_res_both)
     assert_array_equal(both.fit(X_df).transform(X_df), 0.1 * X_res_both)
     assert len(both.transformers_) == 1
-    assert ct.transformers_[-1][0] != 'remainder'
+    assert both.transformers_[-1][0] != 'remainder'
+    assert both.transformers_output_ == {'trans': [0, 1]}
```

Comment:
I would prefer to introduce a different test to only test for `transformers_output_`.